src/lib/components/mods/ModForm.svelte

```diff
@@ -45,14 +46,15 @@
     onSubmit: (submitted: ModData) => onSubmit(trimNonSchema(submitted, modSchema))
   });
 
-  let tags = [];
-  $: {
-    const anyData = $data;
-    if (anyData.tags) {
-      tags = anyData.tags;
-      delete anyData.tags;
-    }
+  let tags = $data.tags;
+  const computeTags = () => {
     $data.tagIDs = tags.map((tag) => tag.id);
+  };
+
+  onMount(computeTags);
```

Comment:
`onMount` is rarely needed, it delays actions to after the DOM has been rendered, rather than on component creation (when the script section runs), so it's normally used when something depends on elements existing in the DOM (for example using a variable with `bind:this`)